### PR TITLE
Implemented decorator getNewAccessTokenUsingRefreshToken

### LINKED ISSUE #27: Support method to simplify token refresh
## 🚀 Feature Proposal

`simple-oauth2` supports a way to easily refresh access tokens by using the `refresh_token` (which you get if you request the `offline` scope:

```javascript
const AccessToken = require('simple-oauth2/lib/access-token')

async function refresh(tokenObject) {
  const token = new AccessToken(tokenObject);
  const refreshedToken = await token.refresh()

  return refreshedToken()
}

const token = {} // ... get token somehow
await refresh(token)
```

It would be cool to expose some shortcut in the fastify plugin to do this, given a OAuth2 token object

## Motivation

We already have a nice `this.getAccessTokenFromAuthorizationCodeFlow(request)` helper function. It would be nice to have something similar for renewing tokens as well.

What do you think?

I am happy to provide an implementation, if this idea makes sense for everyone.

## Example

(see above)

### PR DESCRIPTION
**Pull request by lmammino**
Closes #27 

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -94,11 +94,25 @@ const oauthPlugin = fp(function (fastify, options, next) {
...
+    }
+    getNewAccessTokenWithRefreshTokenCallbacked(refreshToken, params, callback)
+  }
+
   const oauth2 = oauth2Module.create(credentials)
 
   if (startRedirectPath) {
     fastify.get(startRedirectPath, startRedirectHandler)
     fastify.decorate('getAccessTokenFromAuthorizationCodeFlow', getAccessTokenFromAuthorizationCodeFlow)
+    fastify.decorate('getNewAccessTokenWithRefreshToken', getNewAccessTokenWithRefreshToken)
```
Comment (allevo) [inline index.js]: `getNewAccessTokenWithRefreshToken` => `getNewAccessTokenFromRefreshToken` ?

Comment (lmammino) [reply index.js]: Maybe "using" is even more accurate. What do you think?

Comment (allevo) [reply index.js]: Go!

Comment (lmammino) [reply index.js]: Updated and also added some more docs explaining the role of the 2 decorators in the README.

**Comment by allevo** [comment]
published

**Pull request merged**